src/api/route-services/access/index.js

```diff
@@ -0,0 +1,100 @@
+
+const _ = require('lodash');
+
+const config = require('../../../config');
+const AWS = require('../../../utils/requireAWS');
+const getLogger = require('../../../utils/getLogger');
+
+const {
+  createDynamoDbInstance,
+  convertToJsObject,
+  convertToDynamoDbRecord,
+} = require('../../../utils/dynamoDb');
+
+const { isRoleAuthorized } = require('./roles');
+
+const logger = getLogger('[AccessService] - ');
+
+class AccessService {
+  constructor() {
+    this.userAccessTableName = `user-access-${config.clusterEnv}`;
+    this.inviteAccessTableName = `invite-access-${config.clusterEnv}`;
+  }
+
+  // grantRole adds or replaces role for userId and experimentId
+  async grantRole(userId, experimentId, projectId, role) {
+    logger.log(`PUT experiment role ${experimentId} data`);
+
+    const params = {
+      TableName: this.userAccessTableName,
+      Item: {
+        userId,
+        experimentId,
+        projectId,
+        role,
+      },
+    };
+
+    const docClient = new AWS.DynamoDB.DocumentClient({
+      region: config.awsRegion,
+    });
+
+    await docClient.put(params).promise();
+  }
+
+  async canAccessExperiment(userId, experimentId, url, method) {
+    logger.log(`GET experiment permissions ${experimentId} data`);
+    const key = convertToDynamoDbRecord({
+      experimentId,
+      userId,
+    });
+
+    const params = {
+      TableName: this.userAccessTableName,
+      Key: key,
+    };
+
+    const db = createDynamoDbInstance();
+    const response = await db.getItem(params).promise();
+    const { role } = convertToJsObject(response.Item);
```

Comment:
if `response` is none, this line will error out. Can we add some error handling here?

Reply:
no it will not, it will be undefined and be handled correctly by `isRoleAuthorized(...)`

Reply:
so `response` will always have `Item` in it, even if it is empty?

Reply:
no, when there is no Item the call `convertToJsObject(undefined)` will result in `role = undefined` which will be correctly handled by `isRoleAuthorized`. However, I've refactored to make explicit the handling of the `undefined` and avoid confusion in the future.